Core/include/Acts/Geometry/GeometryIdentifier.hpp

```diff
@@ -137,7 +137,10 @@ std::ostream& operator<<(std::ostream& os, GeometryIdentifier id);
 struct GeometryIdentifierHook {
   virtual ~GeometryIdentifierHook() = default;
   virtual Acts::GeometryIdentifier decorateIdentifier(
-      Acts::GeometryIdentifier identifier, const Acts::Surface& surface) const;
+      Acts::GeometryIdentifier identifier,
+      const Acts::Surface& /*surface*/) const {
+    return identifier;
+  }
```

Comment:
it looks odd that we have to move this into the header. do you know why this is necessary? are we missing the compiled source file during linking?

Reply:
This confuses me as well. It's unclear to me why the symbol would not be visible.